src/hubble.rs

```diff
@@ -0,0 +1,250 @@
+        let body = json!({
+                "jsonrpc": "2.0",
+                "method": "hubble_getUserState",
+                "params": [
+                    state_id
+                ],
+                "id": 1u32,
+            }
+        );
+
+        let req = Request::builder()
+            .method(Method::POST)
+            .uri(self.commander_uri.clone())
+            .header(header::CONTENT_TYPE, "application/json")
+            .body(body.to_string().into())?;
+
+        let response = self.client.request(req).await?;
+
+        // asynchronously aggregate the chunks of the body
+        let body = hyper::body::aggregate(response).await?;
+
+        // try to parse as json with serde_json
+        let tx_result: Value = serde_json::from_reader(body.reader())?;
+
+        let user_state = tx_result.get("result").ok_or(Error::HubbleError)?;
+        let user_state = from_str::<UserState>(&user_state.to_string())?;
+        println!("User state {:?}", user_state);
+        Ok(user_state)
+    }
+
+    /// # Errors
+    pub async fn send_create_to_transfer(&self, pub_key: &str) -> EyreResult<String> {
+        let user_state = self.get_user_state(self.from_state_id).await?;
+
+        let body = json!({
+                "jsonrpc": "2.0",
+                "method": "hubble_sendTransaction",
+                "params": [
+                {
+                    "Type": "CREATE2TRANSFER",
+                    "ToPublicKey": pub_key,
+                    "Amount": self.airdrop_amount.to_string(),
+                    "Fee": "1",
+                    "Nonce": user_state.nonce,
+                    "FromStateID": self.from_state_id,
+                    "Signature": "0xABCD0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
+                }
+                ],
+                "id": 1
+            }
+        );
+
+        let req = Request::builder()
+            .method(Method::POST)
+            .uri(self.commander_uri.clone())
+            .header(header::CONTENT_TYPE, "application/json")
+            .body(body.to_string().into())?;
+
+        let response = self.client.request(req).await?;
+
+        // asynchronously aggregate the chunks of the body
+        let body = hyper::body::aggregate(response).await?;
+
+        // try to parse as json with serde_json
+        let tx_result: Value = serde_json::from_reader(body.reader())?;
+
+        let tx_hash = if let Some(tx_hash) = tx_result.get("result") {
+            Ok(tx_hash)
+        } else if let Some(error) = tx_result.get("error") {
+            println!("Error {:?}", error);
+            // TODO report error
+            Err(Error::HubbleError)
+        } else {
+            Err(Error::HubbleError)
+        }?;
+
+        Ok(tx_hash.to_string())
+    }
+
+    /// # Errors
+    pub async fn get_transfer_data(&self, tx_hash: &str) -> EyreResult<CommitmentDetails> {
```

Comment:
We should not use strings for things that are not text. `H256` is a good fit for hashes. Conversion to/from other formats should be done as close to the IO as possible, with appropriate validation.